src/spyglass/common/common_interval.py

```diff
@@ -53,20 +57,144 @@ def insert_from_nwbfile(cls, nwbf, *, nwb_file_name):
-            epoch_dict = {
-                "nwb_file_name": nwb_file_name,
-                "interval_list_name": (
-                    epoch_data.tags[0]
-                    if epoch_data.tags
-                    else f"interval_{epoch_data[0]}"
-                ),
-                "valid_times": np.asarray(
-                    [[epoch_data.start_time, epoch_data.stop_time]]
-                ),
-            }
-
-            cls.insert1(epoch_dict, skip_duplicates=True)
+            inserts.append(
+                {
+                    "nwb_file_name": nwb_file_name,
+                    "interval_list_name": (
+                        epoch_data.tags[0]
+                        if epoch_data.tags
+                        else f"interval_{epoch_data[0]}"
+                    ),
+                    "valid_times": np.asarray(
+                        [[epoch_data.start_time, epoch_data.stop_time]]
+                    ),
+                }
+            )
+
+        cls.insert(inserts, skip_duplicates=True)
+
+    def _generalize_name(self, name):
+        """Generalize interval list name by replacing numbers with '%'."""
+        name = re_sub(r"\d+", "%", name)  # replace numbers with %
+        name = re_sub(r"^%|%$", "", name)  # remove leading/trailing %
+        parts = name.split(".nwb")  # remove file name
+        return parts[-1]
+
+    def _pk_from_key(self, key):
+        return {k: v for k, v in key.items() if k in self.primary_key}
+
+    def key_exists(
+        self,
+        key: dict,
+        approx_name: str = None,
+        time_limit: float = 5.0,
+        *args,
+        **kwargs,
+    ) -> Tuple[bool, dict]:
+        """Check if the interval list already exists.
+
+        Parameters
+        ----------
+        key : dict
+            The key for the IntervalList table.
+        approx_name : str, optional
+            A string that should be in the interval list name. If None, just
+            search by nwb_file_name. Defaults to None. If `True`, will attempt
+            to generalize the name by replacing numbers with '%'.
+        time_limit : float, optional
+            The maximum time to search for an existing interval list before
+            inserting a new one. Defaults to 5.0.
+
+        Returns
+        -------
+        tuple
+            The first element is a boolean indicating if the interval list
+            already exists. The second element is the key of the existing
+            interval list if it exists, or the original key if it does not.
+        """
+
+        start_time = time()
+        query = self & {"nwb_file_name": key.get("nwb_file_name")}
+        if approx_name is True:
+            approx_name = self._generalize_name(key["interval_list_name"])
+        if approx_name:
+            query &= f"interval_list_name LIKE '%{approx_name}%'"
+
+        if not query:
+            return False, key
+
+        candidates = query.fetch(as_dict=True)
+        random_shuffle(candidates)
```

Comment:
Would it make sense to prioritize the ordering of the candidates based on the number parts in their name they share with approx_name?

Reply:
I couldn't think of a good way to do this without a new fuzzy finding dependency.  I suppose the calling tables could specify a delimiter and then items could be ranked by the number of substrings they contain, but I think that would expand the search space more than I'd like. I'd rather keep a narrow scope on the fetch so that pulling the data doesn't slow down the process